pxfish/pyfish.py

```diff
@@ -158,97 +160,130 @@ def open_aquarium_session():
+def get_all_optypes_and_libraries(aq, path):
+    """
+    Retrieves all Operation Types and Libraries 
+
+    Arguments:
+        aq (Session Object): Aquarium session object
+        path (String): the path to where the files will be written
+    """
+    operation_types = aq.OperationType.all()
+    libraries = aq.Library.all()
+    pull(path, operation_types, libraries)
+
+def pull(path, operation_types=[], libraries=[]):
+    """
+    Pulls OperationType and/or Library files from the Aquarium instance.
+
+    Arguments:
+        path (String): the path for the directory where files should be written
+        operation_types (List): list of OperationTypes whose files to pull
+        libraries (List): list of Liraries whose files to pull
+    """
+    for operation_type in operation_types: 
+        write_operation_type(path, operation_type)
 
     for library in libraries:
         write_library(path, library)
 
-def push(directory, category, code_type, op_type_or_library):
-    # make a code object - pull data from json file with aq.Code.new -- and with code data
-    # but contents need to get added in from wherever you saved it
-    aq = open_aquarium_session()
-    # need better way to put this together, but this will do for now.
-    current_directory = directory + "/" + category + "/" + code_type + "/" + op_type_or_library
+def push_library(aq, path, category, library):
+    current_directory = os.path.join(path, category, "libraries", library)
+    push(aq, current_directory, ['source.rb'])
 
-    print(current_directory)
-    #current_directory = os.path.abspath(os.getcwd())
+def push_operation_type(aq, path, category, operation_type):
+    current_directory = os.path.join(path, category, "operation_types", operation_type)
+    push(aq, current_directory, ['cost_model.rb', 'documentation.md', 'precondition.rb', 'protocol.rb'])
 
-    if code_type == library:
-        files_to_write = ['library.rb']
-    else:
-        files_to_write = ['cost_model.rb', 'documentation.md', 'precondition.rb', 'protocol.rb']
-        with open(current_directory + '/definition.json') as f:
-            definitions = json.load(f)
+def push(aq, directory, files_to_write):
+    
+    with open(os.path.join(directory, 'definition.json')) as f:
+        definitions = json.load(f)
 
     for file_name in files_to_write:
-         
-        with open(current_directory + '/' + file_name) as f:
+        with open(os.path.join(directory, file_name)) as f:
              read_file = f.read()
 
         new_code = aq.Code.new(
-                 name=file_name[:-3], # Code Object Name 'protocol, library, cost_model, etc.'
-                 parent_id=definitions['id'], # OperationType or Library id 
-                 parent_class=definitions['parent_class'],   # 'OperationType' or 'Library'
-                 user_id=definitions['user_id'], # User ID from Code object 
-                 content=read_file # Contents of file  
+                 name=file_name[0:-3],
```

Comment:
Use `os.path.splitext` for this instead. See https://docs.python.org/3/library/os.path.html#os.path.splitext